### PR TITLE
GraphQL - ConfigurableCartItem added to Cart returns wrong thumbnail image

### PR DESCRIPTION
**Pull request by shikhamis11**
fixes issue:#28573

If you add a ConfigurableCartItem to the Cart and then Query the Cart, this returns the Thumbnail of the Parent Product instead of the Variant that was selected.

**Query**

```
query CartQuery($cartId: String!) {
  cart(cart_id: $cartId) {
    ...Cart
  }
}

fragment Cart on Cart {
  items {
    id
    quantity
    product {
      id

      thumbnail {
        label
        url
      }
    }
    ... on ConfigurableCartItem {
      configurable_options {
        id
        option_label
        value_label
      }
    }
  }
}
{code}
**Variables**
{code:java}
{
  "cartId": "w74YbL1jbhlrdS95vWQM1mktppa9M3wI"
}
{code}
**Result**
{code:java}
{
  "data": {
    "cart": {
      "items": <
        {
          "id": "241",
          "quantity": 1,
          "product": {
            "id": 1410,
            "thumbnail": {
              "label": "Olivia 1/4 Zip Light Jacket",
              "url": "https://carlos-233-3h6p5qa-a6terwtbk67os.demo.magentosite.cloud/media/catalog/product/cache/ca63cacbf9a8041003a3fd3abbbfe421/w/j/wj12-blue_main.jpg" Should be the Thumbnail for Purple Variant
            }
          },
          "configurable_options": [
            {
              "id": 93,
              "option_label": "Color",
              "value_label": "Purple"
            },
            {
              "id": 162,
              "option_label": "Size",
              "value_label": "XS"
            }
          ]
        }
      >
    }
  }
}

```
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by shikhamis11** [comment]
@magento run all tests

**Comment by shikhamis11** [comment]
@magento run Static Tests

**Comment by shikhamis11** [comment]
@magento run all tests

File: app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/ProductResolver.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['model'])) {
+            throw new LocalizedException(__('"model" value should be specified'));
+        }
+
+        $cartItem = $value['model'];
+        $product = $this->productRepository->get($cartItem->getSku());
```
Comment (cpartica) [inline app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/ProductResolver.php]: this is going to have a big performance problem for loading each product in the cart
it is best to use either data loader which can load all of them (we already do that in catalog)

`Magento\Catalog\Model\Product\Configuration\Item\ItemResolverInterface::getFinalProduct`
Considering that getFinalProduct doesn't add a new query
`$finalProduct = $this->itemResolver->getFinalProduct($cartItem);`

File: app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -61,6 +61,7 @@ input ConfigurableProductCartItemInput {
 type ConfigurableCartItem implements CartItemInterface {
     customizable_options: [SelectedCustomizableOption] @resolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CustomizableOptions")
     configurable_options: [SelectedConfigurableOption!]! @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableCartItemOptions")
+    configured_variant: ProductInterface @doc(description: "Product details of the cart item") @resolver(class: "\\Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ProductResolver")
```
Comment (cpartica) [inline app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls:64]: this also needs tests

**Review by cpartica — CHANGES_REQUESTED** [review]

**Comment by shikhamis11** [comment]
@magento run all tests

**Comment by shikhamis11** [comment]
@magento run all tests

File: app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/ProductResolver.php (hunk 1 above)
Comment (shikhamis11) [reply app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/ProductResolver.php]: I tried with ` $finalProduct = $this->itemResolver->getFinalProduct($cartItem);`  
This provides all information related parent Items only same as `CartItemInterface.product` object except `ProductInerface.thumbnail` object as per set on backend configuration ` checkout/cart/configurable_product_image` either `parent` or `itself`

**Comment by shikhamis11** [comment]
Closing this as there is another PR

**Pull request closed without merging**